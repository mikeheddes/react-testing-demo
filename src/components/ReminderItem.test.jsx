import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import ToDoItem from "./ReminderItem";

describe("<ReminderItem />", () => {
  it("match snapshot", () => {
    const onRemove = jest.fn();
    const tree = renderer.create(
      <ToDoItem onRemove={onRemove}>The Label</ToDoItem>
    );
    expect(tree).toMatchSnapshot();
  });
});
