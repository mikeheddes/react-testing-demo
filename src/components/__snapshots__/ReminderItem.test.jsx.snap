// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReminderItem /> match snapshot 1`] = `
.c2 {
  width: 5px;
  background-color: white;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 {
  background-color: #495265;
  position: relative;
  box-shadow: 8px 8px rgba(0,0,0,0.15);
  margin: 30px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  display: block;
  text-align: center;
  color: white;
  font-weight: 500;
  font-size: 42px;
  padding: 39px 50px;
  padding: 20px 30px;
  font-size: 34px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  text-align: left;
}

.c3 {
  color: white;
  font-weight: 500;
  font-size: 42px;
  padding: 39px 50px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  outline: none;
  background-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  padding: 20px 30px;
  font-size: 34px;
}

.c3:active {
  opacity: 0.5;
}

.c4 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  display: inline-block;
}

<div
  className="c0"
  color="#495265"
>
  <span
    className="c1"
  >
    The Label
  </span>
  <div
    className="c2"
  />
  <button
    className="c3"
    onClick={[MockFunction]}
  >
    <span
      className="c4"
    >
      +
    </span>
  </button>
</div>
`;
